src/Core/Checkout/Promotion/PromotionException.php

```diff
@@ -200,4 +197,65 @@ public static function promotionSetGroupNotFound(string $groupId): self
             ['id' => $groupId],
         );
     }
+
+    public static function missingRequestParameter(string $name): self
+    {
+        return new self(
+            Response::HTTP_BAD_REQUEST,
+            self::MISSING_REQUEST_PARAMETER_CODE,
+            'Parameter "{{ parameterName }}" is missing.',
+            ['parameterName' => $name]
+        );
+    }
+
+    /**
+     * @deprecated tag:v6.8.0 - reason:return-type-change - Will return self
+     */
+    public static function priceNotFound(LineItem $lineItem): self|PriceNotFoundException
```

Comment:
as `PriceNotFoundException` is now extending this domain exception, you should already be able to use self only